lib/internal/Magento/Framework/App/DeploymentConfig/Reader.php

```diff
@@ -81,47 +81,57 @@ public function getFiles()
     }
 
     /**
-     * Method loads merged configuration within all configuration files.
-     * To retrieve specific file configuration, use FileReader.
-     * $fileKey option is deprecated since version 2.2.0.
-     *
-     * @param string $fileKey The file key (deprecated)
-     * @return array
-     * @throws FileSystemException If file can not be read
-     * @throws RuntimeException If file is invalid
-     * @throws \Exception If file key is not correct
-     * @see FileReader
+     * @param $fileKey
+     * @return array|mixed
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\RuntimeException
      */
     public function load($fileKey = null)
     {
+        static $cache = [];
+
         $path = $this->dirList->getPath(DirectoryList::CONFIG);
         $fileDriver = $this->driverPool->getDriver(DriverPool::FILE);
         $result = [];
+
         if ($fileKey) {
             $filePath = $path . '/' . $this->configFilePool->getPath($fileKey);
-            if ($fileDriver->isExists($filePath)) {
-                $result = include $filePath;
-                if (!is_array($result)) {
-                    throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$filePath]));
-                }
-            }
+            $cache = $this->getFileCache($cache, $filePath, $fileDriver);
+            $result = $cache[$filePath] ?: [];
         } else {
             $configFiles = $this->getFiles();
             foreach ($configFiles as $file) {
                 $configFile = $path . '/' . $file;
-                if ($fileDriver->isExists($configFile)) {
-                    $fileData = include $configFile;
-                    if (!is_array($fileData)) {
-                        throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$configFile]));
-                    }
-                } else {
-                    continue;
-                }
-                if ($fileData) {
-                    $result = array_replace_recursive($result, $fileData);
+                $cache = $this->getFileCache($cache, $configFile, $fileDriver);
+                if ($cache[$configFile]) {
+                    $result = array_replace_recursive($result, $cache[$configFile]);
                 }
             }
         }
         return $result ?: [];
     }
+
+    /**
+     * @param array $cache
+     * @param string $filePath
+     * @param \Magento\Framework\Filesystem\DriverInterface $fileDriver
+     * @return array
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\RuntimeException
+     */
+    public function getFileCache(array $cache, string $filePath, \Magento\Framework\Filesystem\DriverInterface $fileDriver): array
+    {
```

Comment:
It appears that the `getFileCache` method is only used within the Reader class, it should be declared as a private scope.

It appears that the `getFileCache` does not throw \Magento\Framework\Exception\FileSystemException. Can you confirm this?